### PR TITLE
Fix bug in faster interval rounding

### LINKED ISSUE #56400: [CI] testRandomTimeIntervalRounding Failure
**Build scan**: https://gradle-enterprise.elastic.co/s/plit7g4fdojhi/tests/

**Repro line**: 
```
./gradlew ':server:test' \
--tests "org.elasticsearch.common.RoundingTests.testRandomTimeIntervalRounding" \
-Dtests.seed=DE09A1CC10383EB0 -Dtests.security.manager=true \
-Dtests.locale=ca-ES -Dtests.timezone=Etc/GMT+1 \
-Dcompiler.java=14 -Druntime.java=11
```

**Reproduces locally?**: No

**Applicable branches**: master

**Failure excerpt**:
```
Expected [148 MINUTES in America/Menominee] to \
 round [2008-11-02T07:50:16.470Z] \
    to [2008-11-02T06:56:00Z] but instead rounded \
    to [2008-11-02T07:56:00Z]
at __randomizedtesting.SeedInfo.seed([DE09A1CC10383EB0:7D4B1688B8C503D4]:0)
at org.junit.Assert.fail(Assert.java:88)
at org.elasticsearch.common.RoundingTests.testRandomTimeIntervalRounding(RoundingTests.java:410)
```

### PR DESCRIPTION
**Pull request by nik9000**
This fixes a bug in the interval rounding and two test bugs that showed
up when I ran 1000s of iterations of the tests. The bug was that we
could end up with duplicate transitions if we only needed the last
transition from the list of fully defined transitions and the
transitions overlapped with the rules. This happens. Its ok. We had
defence against that if we needed more than one transition but forgot it
in this case. Now we have it and assertions to make sure we catch any
similar mistakes.

One test bug had to do with the randomized test calling
`round(round(min))` because sometimes the first `round` call will return
a time less than min. Specifically if `min` is near daylight savings
time. Anyway, this change fixes it by building an extra-wide prepared
rounding just in case.

The other test bug came up when adding a test for the real bug, namely
that `assertRoundingAtOffset` made an assertion that wasn't true if the
time to round ended up being in an overlap. This just drops that
assertion. We have similar assertions in cases where we *know* what kind
of offsets we're working with in other tests.

Closes #56400

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/common/LocalTimeOffset.java [hunk 1]
```diff
@@ -560,6 +563,11 @@ private static LocalTimeOffset checkForFixedZone(ZoneId zone, ZoneRules rules) {
         if (false == itr.hasNext()) {
             if (minSecond < t.toEpochSecond() && t.toEpochSecond() < maxSecond) {
                 transitions.add(t);
+                /*
+                 * Sometimes the rules duplicate the transitions. And
+                 * duplicates confuse us. So we have to skip past them.
+                 */
+                minSecond = t.toEpochSecond() + 1;
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/common/LocalTimeOffset.java:570]: This is the actual bug fix.

**Comment by nik9000** [comment]
run elasticsearch-ci/docs

File: server/src/test/java/org/elasticsearch/common/LocalTimeOffsetTests.java [hunk 2]
```diff
@@ -155,6 +154,22 @@ public void testLastTransitionWithoutRules() {
         assertRoundingAtOffset(lookup.lookup(time), time, TimeUnit.MINUTES.toMillis(345));
     }
 
+    public void testLastTransitionOverlapsRules() {
+        /*
+         * America/Tijuana's transitions overlap its rules and we have to make
```
Comment (not-napoleon) [inline server/src/test/java/org/elasticsearch/common/LocalTimeOffsetTests.java:159]: I don't understand the phrase "transitions overlap its rules"

**Review by not-napoleon — APPROVED** [review]
LGTM.

**Comment by nik9000** [comment]
run elasticsearch-ci/packaging-sample-unix-docker

**Comment by nik9000** [comment]
I'm backporting this in #56396.

**Pull request merged**